addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/protocol/xml/InputConverterXML.java

```diff
@@ -0,0 +1,167 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.yamahareceiver.internal.protocol.xml;
+
+import org.apache.commons.lang.StringUtils;
+import org.openhab.binding.yamahareceiver.internal.protocol.AbstractConnection;
+import org.openhab.binding.yamahareceiver.internal.protocol.InputConverter;
+import org.openhab.binding.yamahareceiver.internal.protocol.ReceivedMessageParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.lang.ref.WeakReference;
+import java.util.*;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.toSet;
+import static org.openhab.binding.yamahareceiver.YamahaReceiverBindingConstants.*;
+
+/**
+ * XML implementation of {@link InputConverter}.
+ *
+ * @author Tomasz Maruszak - Input mapping fix
+ *
+ */
+public class InputConverterXML implements InputConverter {
+
+    private final Logger logger = LoggerFactory.getLogger(InputConverterXML.class);
+
+    private final WeakReference<AbstractConnection> comReference;
+
+    /**
+     * User defined mapping for state to input name.
+     */
+    private final Map<String, String> inputMap;
+    /**
+     * Holds all the inputs names that should NOT be transformed by the {@link #convertNameToID(String)} method.
+     */
+    private final Set<String> inputsWithoutMapping;
+
+    public InputConverterXML(AbstractConnection con, String inputMapConfig) {
+        this.comReference = new WeakReference<>(con);
+
+        logger.trace("User defined mapping: {}", inputMapConfig);
+        this.inputMap = createMapFromSetting(inputMapConfig);
+
+        try {
+            this.inputsWithoutMapping = createInputsWithoutMapping();
+            logger.trace("These inputs will not be mapped: {}", inputsWithoutMapping);
+        } catch (IOException | ReceivedMessageParseException e) {
+            throw new RuntimeException("Could not communicate with the device", e);
+        }
+    }
+
+    /**
+     * Creates a map from a string representation: "KEY1=VALUE1,KEY2=VALUE2"
+     *
+     * @param setting
+     * @return
+     */
+    private Map<String, String> createMapFromSetting(String setting) {
```

Comment:
Hm. I don't like this. If the framework does not support a valid use-case, the framework should be extended. This is clearly support code to workaround the limitation of missing list-like configuration entries.

Reply:
By saying the framework, you mean ESH?

Reply:
Yes I do. An ESH implementation could take some time and I don't want to block this PR, so this is not required to be changed right now :)